src/python/gudhi/delaunay_complex.pyx

```diff
@@ -103,25 +103,28 @@ cdef class DelaunayComplex:
          """
         return self.this_ptr != NULL
 
-    def create_simplex_tree(self, max_alpha_square = float('inf'), filtration = None):
+    def create_simplex_tree(self, max_alpha_square: float = float('inf'),
+                            filtration: Optional[Literal['alpha', 'cech']] = None,
+                            square_root_filtrations: bool = False):
```

Comment:
I didn't know (or remember) that this was possible, but it looks good, there are probably more places where we could simplify the code this way. We need to check what the generated documentation looks like (I think we want to see the Python types there).

Reply:
Sphinx was not ok with your proposal (the type was missing), so I go for the very redundant following syntax:
```python
    def create_simplex_tree(self, double max_alpha_square:float = float('inf'),
                            filtration: Optional[Literal['alpha', 'cech']] = None,
                            bool square_root_filtrations:bool = False) -> SimplexTree:
```
on dc78dc15b

Reply:
I don't remember : does gudhi uses [embedsignature](https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives) ? This may solve this redundant issue.